### PR TITLE
[IMPROVEMENT] E2E appium/app launch times

### PR DESCRIPTION
**Pull request by SamuelSalas**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: wdio/config/android.config.debug.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ config.capabilities = [
     noReset: false,
     fullReset: false,
     maxInstances: 1,
-    deviceName: 'Android 11 - Pixel 4a API 3',
-    platformVersion: '11',
+    deviceName: 'R5CRA0C5ZXW',
```
Comment (NicolasMassart) [inline wdio/config/android.config.debug.js:11]: Was this expected to be committed?

Comment (SamuelSalas) [reply wdio/config/android.config.debug.js]: Already commit the proper changes, thanks for point this out

**Review by chrisleewilcox — APPROVED** [review]

**Review by cortisiko — CHANGES_REQUESTED** [review]

File: wdio/step-definitions/app-launch-times.steps.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+When(/^I kill then app$/, async () => {
+  await driver.closeApp();
+});
+
+When(/^I relaunch the app$/, async () => {
+  await driver.startActivity('io.metamask.qa', 'io.metamask.MainActivity');
+  startTimer = new Date().getTime();
+});
+
+When(/^I log into my wallet$/, async () => {
```
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/app-launch-times.steps.js:30]: I dont see a common step for login once the app has been close in the same session

File: wdio/step-definitions/app-launch-times.steps.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+Given(/^the app is closed$/, async () => {
+  startTimer = new Date().getTime();
+});
+
+When(/^I launch the app$/, async () => {
+  await WelcomeScreen.waitForSplashAnimationToDisplay();
+});
+
+Given(/^I have imported my wallet for app launch times$/, async () => {
```
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/app-launch-times.steps.js:53]: The main reason was to both remove one hard coded driver.pause() and second one is that there while the steps are pretty much similar, i needed to move the wait for splash animation to appear to a different step so i could stop the timer

File: wdio/step-definitions/common-steps.js [hunk 4]
```diff
@@ -22,6 +22,7 @@ Given(/^I have imported my wallet$/, async () => {
   await MetaMetricsScreen.isScreenTitleVisible();
   await MetaMetricsScreen.tapIAgreeButton();
   await ImportFromSeedScreen.isScreenTitleVisible();
+  await ImportFromSeedScreen.setLoginBiometrics('OFF');
```
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/common-steps.js:25]: Remove

File: wdio/step-definitions/app-launch-times.steps.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  await driver.closeApp();
+});
+
+When(/^I relaunch the app$/, async () => {
+  await driver.startActivity('io.metamask.qa', 'io.metamask.MainActivity');
+  startTimer = new Date().getTime();
+});
+
+When(/^I log into my wallet$/, async () => {
+  await WelcomeScreen.isScreenTitleVisible();
```
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/app-launch-times.steps.js:31]: My bad. The waitForSplashAnimationToDisplay() method has a hard coded pause, i created another method that just wait for the splash animation instead. Going to replace it with just the wait for the splash animation.

File: wdio/features/AppLaunchTime.feature [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  Scenario: Measure app launch time on fresh install
+    Given the app is closed
+    When I launch the app
+    Then the app should launch within x seconds
+
+  Scenario: Measure app launch time after importing a wallet
+    Given I have imported my wallet for app launch times
+    And I tap No Thanks on the Enable security check screen
+    And I tap No thanks on the onboarding welcome tutorial
+    And main Wallet Screen is displayed
```
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/features/AppLaunchTime.feature:16]: updated

File: wdio/step-definitions/app-launch-times.steps.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import WelcomeScreen from '../screen-objects/Onboarding/OnboardingCarousel';
+import OnboardingScreen from '../screen-objects/Onboarding/OnboardingScreen';
+import MetaMetricsScreen from '../screen-objects/Onboarding/MetaMetricsScreen';
+import ImportFromSeedScreen from '../screen-objects/Onboarding/ImportFromSeedScreen';
+
+let startTimer;
+let stopTimer;
+const validAccount = Accounts.getValidAccount();
+
+Given(/^main Wallet Screen is displayed$/, async () => {
```
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/app-launch-times.steps.js:17]: remove

File: wdio/step-definitions/app-launch-times.steps.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+});
+
+Then(/^the app should launch within x seconds$/, async () => {
+  stopTimer = new Date().getTime();
+  const result = stopTimer - startTimer;
+  cucumberJson.attach(`Milliseconds: ${result}`);
+});
+
+Given(/^the app is closed$/, async () => {
+  startTimer = new Date().getTime();
```
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/app-launch-times.steps.js:46]: Update the gherkin for that step definition

[the remark this answers is not in the document]
Comment (cortisiko) [reply wdio/step-definitions/app-launch-times.steps.js:39]: okay updated.

File: wdio/step-definitions/app-launch-times.steps.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (cortisiko) [reply wdio/step-definitions/app-launch-times.steps.js]: trigger the test against Browserstack and see what the test runs look like

File: wdio/step-definitions/app-launch-times.steps.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/app-launch-times.steps.js:23]: Got it

File: wdio/step-definitions/app-launch-times.steps.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (SamuelSalas) [reply wdio/step-definitions/app-launch-times.steps.js]: Got it

**Comment by SamuelSalas** [comment]
Browerstack successful runs:
https://app-automate.browserstack.com/dashboard/v2/builds/af352c3cb4fc61986127b52d3b21c37a2213b608/sessions/b4b7f35b875b6dbcbee34304012c7e4ebef4db2e?searchQuery=b4b7f35b875b6dbcbee34304012c7e4ebef4db2e

**Comment by SamuelSalas** [comment]
https://app-automate.browserstack.com/dashboard/v2/builds/af352c3cb4fc61986127b52d3b21c37a2213b608/sessions/45336fe6a5923b8ba7aa5cd62e840e27098ad893?searchQuery=45336fe6a5923b8ba7aa5cd62e840e27098ad893

**Comment by SamuelSalas** [comment]
https://app-automate.browserstack.com/dashboard/v2/builds/af352c3cb4fc61986127b52d3b21c37a2213b608/sessions/442db6ff5fc55c5ba10563d147b1cd3f5d8f6082?searchQuery=442db6ff5fc55c5ba10563d147b1cd3f5d8f6082

**Review by cortisiko — APPROVED** [review]
great stuff @SamuelSalas 👏 👏

**Pull request merged**